crates/data_components/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#![allow(clippy::missing_errors_doc)]
+use std::{error::Error, sync::Arc};
+
+use async_trait::async_trait;
+use datafusion::{common::OwnedTableReference, datasource::TableProvider};
+
+pub mod arrow;
+pub mod databricks;
+
+#[async_trait]
+pub trait Read {
```

Comment:
We need to work with trait objects for `TableProvider`s because that is what a lot of the DataFusion APIs accept, and also we want to eventually allow extending Spice using a plugin system, which would require trait objects - so that is why you might see us use trait objects more than is strictly required.